src/NCronJob/Scheduler/JobQueueManager.cs

```diff
@@ -53,7 +57,11 @@ public void RemoveQueue(string queueName)
                 jobQueue.Clear();
                 jobQueue.CollectionChanged -= CallCollectionChanged;
                 semaphores.TryRemove(queueName, out _);
-                jobCancellationTokens.Clear();
+                if (jobCancellationTokens.TryRemove(queueName, out CancellationTokenSource? x))
+                {
+                    x.Cancel();
```

Comment:
Dropping this make some RuntimeRegistry tests hang forever

Reply:
Interesting - your approach is the correct one. Clearing everything was not the best move